### PR TITLE
[SECURITY-334] Hard reset link must be also protected

### PR DESCRIPTION
**Pull request by Wadeck**
- provide the form that go with it
- correct the redirect that does not take into consideration the context

[SECURITY-334](https://issues.jenkins-ci.org/browse/SECURITY-334) (private security issue)

@reviewbybees @imod @daniel-beck

### CODE REVIEW

File: src/main/java/org/jenkinsci/plugins/scriptler/ScriptlerManagement.java [hunk 1]
```diff
@@ -303,10 +303,11 @@ private GitScriptlerRepository getGitRepo() {
      * @return redirects to the repo entry page at <code>http://jenkins.orga.com/scriptler.git</code>
      * @throws IOException
      */
+    @RequirePOST
     public HttpResponse doHardResetGit() throws IOException {
         checkPermission(Hudson.ADMINISTER);
         getGitRepo().hardReset();
-        return new HttpRedirect("/scriptler.git");
+        return new HttpRedirect("../scriptler.git");
```
Comment (daniel-beck) [inline src/main/java/org/jenkinsci/plugins/scriptler/ScriptlerManagement.java]: Unclear why this is needed, was the direct broken before?

Comment (Wadeck) [reply src/main/java/org/jenkinsci/plugins/scriptler/ScriptlerManagement.java]: If you have a context like "jenkins", so the redirect will put you from `http://localhost:8080/jenkins/scriptler.git/` to `http://localhost:8080/scriptler.git`

Comment (daniel-beck) [reply src/main/java/org/jenkinsci/plugins/scriptler/ScriptlerManagement.java]: Ah, I missed that there was a leading `/`. Now it's clear.

**Comment by Wadeck** [comment]
seems @reviewbybees done

**Pull request merged**